presto-router/src/main/java/com/facebook/presto/router/cluster/ClusterManager.java

```diff
@@ -207,6 +213,22 @@ public Optional<URI> getDestination(RequestInfo requestInfo)
         if (config.getSchedulerType() == WEIGHTED_RANDOM_CHOICE || config.getSchedulerType() == WEIGHTED_ROUND_ROBIN) {
             config.getScheduler().setWeights(config.getServerWeights().get(groupSpec.getName()));
         }
+        else if (config.getSchedulerType() == CUSTOM_PLUGIN_SCHEDULER) {
+            try {
+                //Set remote cluster infos in the custom plugin scheduler
+                List<URI> finalHealthyClusterURIs = healthyClusterURIs;
+                Map<URI, ClusterInfo> healthyRemoteClusterInfos = remoteClusterInfos.entrySet().stream()
+                        .filter(entry -> finalHealthyClusterURIs.contains(entry.getKey()))
```

Comment:
I don't really understand why we need this filter? Why don't we just iterate through the entries in the existing Map and filter out the cluster infos which are not healthy?

Reply:
Also, I think we can make this more optimized here. Instead of having to generate a whole map, I think we can just use `Maps.filterValues` from guava. It returns a view over a map which filters out entries whose values don't satisfy a predicate. In this case it could just be `RemoteClusterInfo::isHealthy`.

`Maps.filterValues` returns a view which reflects modifications to the underlying map. If we're worried about updates occurring to the Map, we can always do a single copy of it with `ImmutableMap.copyOf()` to defend against this. I just don't like the fact we're potentially doing multiple O(n) searches on the list for `.contains` here when it's unnecessary